packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -153,6 +158,47 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     }
   }
 
+  function commitBeforeMutationLifeCycles(
+    current: Fiber | null,
+    finishedWork: Fiber,
+  ): void {
+    switch (finishedWork.tag) {
+      case ClassComponent: {
+        const instance = finishedWork.stateNode;
+        if (finishedWork.effectTag & Snapshot) {
+          if (current !== null) {
+            const prevProps = current.memoizedProps;
+            const prevState = current.memoizedState;
+            startPhaseTimer(finishedWork, 'getSnapshotBeforeUpdate');
+            instance.props = finishedWork.memoizedProps;
+            instance.state = finishedWork.memoizedState;
+            const snapshot = instance.getSnapshotBeforeUpdate(
+              prevProps,
+              prevState,
+            );
+            // TODO Warn about undefined return value
+            current.snapshot = snapshot != null ? snapshot : null;
```

Comment:
What's the point of this null check?

Reply:
To pass `null` instead of `undefined`

Reply:
Why though

Reply:
Seems better to pass a consistent value regardless of whether `getSnapshotBeforeUpdate` exists or not.

Reply:
Let's stop this mini review for now 😄 I don't think it's useful. I'll let you know when the PR is ready.